docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+by FRR. OVN-Kubernetes will manage FRR configuration through the FRR-K8S API in order to advertise routes outside of the
+node onto the provider network.
+
+There should be no changes required in FRR. FRR-K8S may need extended APIs to cover the OVN-Kubernetes use cases
+proposed in this enhancement. OVN will require no changes.
+
+### Workflow Description
+
+An admin has the ability to configure BGP peering and choose what networks to advertise. A tenant is able to define
+networks for their namespace, but requires admin permission in order to expose those networks onto the provider's BGP
+fabric. A typical workflow will be for a user or admin to create a user-defined network, and then the admin will be
+responsible to:
+
+1. If setting up VRF-Lite, do any host modifications necessary via NMState to enslave an IP interface to the
+   matching network VRF.
+2. Configure BGP peering via interacting with the FRR-K8S API for a given set of worker nodes. Also define filters for
+   what routes should be received from the provider network.
+3. Create the OVN-Kubernetes RouteAdvertisements CR to configure what routes and where to advertise them.
+4. Verify peering and that routes have been propagated correctly.
+5. If desired, change the transport type from geneve on the network to none in order to no longer use an overlay.
+
+For detailed examples, see the [BGP Configuration](#bgp-configuration) section.
+
+### API Extensions
+FRR-K8S API will be used in order to create BGP Peering and configure other BGP related configuration. A
+RouteAdvertisements CRD will be introduced in order to determine which routes should be advertised for specific networks.
+Additionally, the network CRD will be modified in order to expose a new transport field to determine if encapsulation
+should be used for east/west traffic.
+
+### API Details
+
+#### Route Advertisements
+
+When OVN-Kubernetes detects that a FRR-K8S CR has been created for BGP peering, OVN-Kubernetes will by default
+advertise the pod subnet for each node, via creating an additive, per node FRR-K8S CR that is managed by OVN-Kubernetes.
+OVN-Kubernetes CRD:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+    # networkSelector:
+    # nodeSelector:
+    # frrConfigurationSelector:
+    targetVRF: default
+    advertisements:
+      podNetwork: true
+      egressIP: true
+```      
+
+In the above example, an optional networkSelector may also be optionally supplied which will match namespaces/networks.
+When omitted, RouteAdvertisements will be applied to the default cluster network. If a selector is used that selects the
+default cluster network, it will be enabled for any namespace using the cluster default network. Additionally,
+advertisements may be limited to specific nodes via the nodeSelector.
+
+A networkSelector may select more than just the cluster default network, including user-defined networks. In such a case
```

Comment:
should we remove UDNs from scope? or we keep things as is in the OKEP and its OK if some things are not implemented yet I gues?